### PR TITLE
WL-4002 Multiple files can be added to a Lesson's page in one go.

### PR DESCRIPTION
**Pull request by ouit0408**
Overrid the CommonsMultipartResolver's parseFileItems method to enable multiple
file upload from one form input type.

### CODE REVIEW

**Comment by buckett** [comment]
I think you're IDE trimmed all trailing whitespace when saving one of the files. In general you shouldn't have trailing whitespace on any lines that you write, but it looks like you've cleaned up other peoples code as well. This makes the code harder to patch against Sakai as it's much more likely that one of the lines you've cleaned up will have been changed in Sakai and so cause a conflict when applying the patch.

**Comment by buckett** [comment]
Ah, it's only the method you edited that get cleaned up.

File: tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java [hunk 1]
```diff
@@ -5693,8 +5703,8 @@ else if (multimediaDisplayType == 3)
 		} finally {
```
Comment (buckett) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: The `securityService.pushAdvisor()` is in a different method which I think will result in the advisor being present for the first file, but not any others.

File: tool/src/java/org/sakaiproject/lessonbuildertool/resolver/MultiCommonsMultipartResolver.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by neelam on 8/3/2015.
+ * This class extends CommonsMultipartResolver's parseFileItems() method to upload multiple files coming from same form input name.
+ * Linked to JIRA WL-4002
```
Comment (buckett) [inline tool/src/java/org/sakaiproject/lessonbuildertool/resolver/MultiCommonsMultipartResolver.java]: Don't put WL-refs in the comments as we will send patches like this back to Sakai and they won't mean anything there.

File: tool/src/webapp/WEB-INF/requestContext.xml [hunk 3]
```diff
@@ -248,6 +249,10 @@
 		<property name="lessonsAccess" ref="org.sakaiproject.lessonbuildertool.service.LessonsAccess"/>
 		<property name="lessonBuilderAccessService" ref="org.sakaiproject.lessonbuildertool.service.LessonBuilderAccessService" />
 	</bean>
+	<!-- Over-riding the CommonsMultipartResolver to have multiple files incoming under the same form input name  -->
+	<bean id="multipartResolver" class="org.sakaiproject.lessonbuildertool.resolver.MultiCommonsMultipartResolver">
+		<property name="maxUploadSize" value="100000"/>
```
Comment (buckett) [inline tool/src/webapp/WEB-INF/requestContext.xml]: The maxUploadSize should come from the ServerConfigurationService: `content.upload.max` I think.

Comment (ouit0408) [inline tool/src/webapp/WEB-INF/requestContext.xml]: as content size is checked in the method(uploadSizeOk) , dropping this property from here.

**Comment by ouit0408** [comment]
Implemented the review comments and have pushed the changes.

File: tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java [hunk 4]
```diff
@@ -5436,37 +5440,43 @@ public void addMultimedia() {
...
+		if (!itemOk(itemId))
+			return;
+		if (!canEditPage())
+			return;
+		if (!checkCsrf())
+			return;
+
+		if (multipartMap.size() > 0) {
+			// 	user specified a file, create it
+			Iterator iterator = multipartMap.values().iterator();
```
Comment (buckett) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: Generics? Java 5 for loop?

Comment (buckett) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: Sorry for not spotting this before.

Comment (ouit0408) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: Added enhanced for loop and have pushed the changes.

File: tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java [hunk 5]
```diff
@@ -5436,37 +5440,40 @@ public void addMultimedia() {
...
+				addMultimediaFile(file);
 			}
-			if (!itemOk(itemId))
-				return;
-			if (!canEditPage())
-				return;
-			if (!checkCsrf())
-			    return;
+		}
+		if(advisor != null) securityService.popAdvisor();
```
Comment (buckett) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: This must be in a finally block and the pushAdvisor must be in the try.

Comment (ouit0408) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: Changes done and code pushed.

File: tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java [hunk 6]
```diff
@@ -253,6 +254,7 @@
 	private String currentSiteId = null;
 
 	public Map<String, MultipartFile> multipartMap;
+	private MultipartResolver multipartResolver;
```
Comment (buckett) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: This is never read, why do you need it?

Comment (ouit0408) [inline tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java]: Deleted and pushed the code.

**Pull request merged**